app/usedgoodreads/__main__.py

```diff
@@ -15,32 +15,59 @@ def index(key):
     cache = Book.get(key=key)
 
     if cache:
+        # Book is in database
         books = UsedBook.get(isbn=cache.isbn)
-
         if books:
+            # Used books are in database
             results = [{"title": v.title, "description": v.description,
                         "price": v.price, "link": v.link} for v in books]
 
             results.sort(key=lambda each: each["price"])
 
-            return render_template("index.html", isbn=cache.isbn, title=cache.title,
+            return render_template("index.html", isbn=cache.isbn,
+                                   title=cache.title,
                                    results=results)
-
-    ticket = Ticket(key)
-
-    if q.fetch_job(ticket.isbn_jid):
-        return render_template("index.html")
-
-    if q.fetch_job(ticket.used_books_jid):
-        return render_template("index.html", status="searching")  # TODO: Should return actual status
-
-    q.enqueue(resolve_goodreads_key, ticket, job_timeout=60 * 1,
-            ttl=60 * 60, job_id=ticket.isbn_jid)
-
-    q.enqueue(resolve_used_books, key, job_timeout=60 * 1,
-              ttl=60 * 60, job_id=ticket.used_books_jid, depends_on=ticket.isbn_jid)
-
-    return render_template("index.html")
+        else:
+            # Used books need to be fetched
+            ticket = Ticket(key)
+            job = q.fetch_job(ticket.used_books_jid)
+            if job:
+                # Fetch job already in q
+                if job.get_status() == "finished":
```

Comment:
What I don't like here is rq only keeps the results for a specific time. This means our responses here change based on if the rq job is still in the queue or no longer in the queue. :thinking:

Reply:
Having a `ttl` for the results is actually pretty need, because you can set it up as a timeout for when you want to search again. If it is no longer in the queue that means we decided the last search is too long ago. And search again. No?

Reply:
Gotcha, we're using redis for persistence now, too. Not sure how I feel about this, but more on that in other comments.